File: test/python/test_device.py
#!/usr/bin/env python3
"""test_device.py — 장치 연결 / 기본 정보 테스트

Usage: python test_device.py [ip] [port]
"""
import sys
from prism import Prism, PrismError

_pass = 0
_fail = 0

def check(num: int, name: str, cond: bool, detail: str = "") -> None:
    global _pass, _fail
    if cond:
        _pass += 1;  tag = "[PASS]"
    else:
        _fail += 1;  tag = "[FAIL]"
    print(f"{tag} {num:02d} {name}" + (f" : {detail}" if detail else ""))


def main(ip: str, port: int) -> int:
    prism = Prism(ip, port)

    # 01  open
    prism.open()
    check(1, "open", prism.is_open())

    # 02  fw_version
    ver = prism.fw_version()
    check(2, "fw_version", len(ver) > 0, f'"{ver}"')

    # 03  serial
    sn = prism.serial()
    check(3, "serial", len(sn) > 0, f'"{sn}"')

    # 04  info
    inf    = prism.info()
    ok_key = all(k in inf for k in ("num_ai_channels", "ai_min_code",
                                    "ai_max_code", "ai_min_voltage", "ai_max_voltage"))
    ok_rng = inf.get("ai_min_code", 0) < inf.get("ai_max_code", 0)
    detail = (f"num_ch={inf.get('num_ai_channels')}  "
              f"min_code={inf.get('ai_min_code')}  "
              f"max_code={inf.get('ai_max_code')}")
    check(4, "info", ok_key and ok_rng, detail)

    # 05  close
    prism.close()
    check(5, "close", not prism.is_open())

    total = _pass + _fail
    print(f"\nResults: {_pass}/{total} passed")
    return 0 if _fail == 0 else 1


if __name__ == "__main__":
    ip   = sys.argv[1] if len(sys.argv) > 1 else "192.168.7.1"
    port = int(sys.argv[2]) if len(sys.argv) > 2 else 7777
    sys.exit(main(ip, port))

File: test/c/test_device.c
/*
 * test_device.c — 장치 연결 / 기본 정보 테스트
 * Build: make test
 * Run  : ./test/c/test_device 192.168.7.1 7777
 */
#include "prismc.h"
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

static int g_pass = 0, g_fail = 0;

static void check(int num, const char *name, int cond, const char *detail)
{
    if (cond) { g_pass++; printf("[PASS] %02d %-35s", num, name); }
    else       { g_fail++; printf("[FAIL] %02d %-35s", num, name); }
    if (detail && detail[0]) printf(": %s", detail);
    printf("\n");
}

int main(int argc, char *argv[])
{
    const char *ip   = (argc > 1) ? argv[1] : "192.168.7.1";
    uint16_t    port = (argc > 2) ? (uint16_t)atoi(argv[2]) : 7777;

    /* 01  open */
    prismc_t *dev = prismc_open(ip, port);
    check(1, "open", dev != NULL, "");

    if (!dev) {
        printf("\nResults: 0/1 passed\n");
        return 1;
    }

    /* 02  fw_version */
    char ver[32] = {0};
    int rc = prismc_fw_version(dev, ver);
    check(2, "fw_version", rc == RESULT_SUCCESS && strlen(ver) > 0, ver);

    /* 03  serial */
    char sn[32] = {0};
    rc = prismc_serial(dev, sn);
    check(3, "serial", rc == RESULT_SUCCESS && strlen(sn) > 0, sn);

    /* 04  info */
    struct PrismcDeviceInfo *inf = prismc_info(dev);
    int ok = (inf != NULL)
          && (inf->AI_MIN_CODE < inf->AI_MAX_CODE)
          && (inf->NUM_AI_CHANNELS > 0);
    char detail[64] = {0};
    if (inf)
        snprintf(detail, sizeof(detail), "num_ch=%u  min_code=%d  max_code=%d",
                 inf->NUM_AI_CHANNELS, inf->AI_MIN_CODE, inf->AI_MAX_CODE);
    check(4, "info", ok, detail);

    /* 05  close */
    rc = prismc_close(dev);
    check(5, "close", rc == RESULT_SUCCESS, "");

    printf("\nResults: %d/%d passed\n", g_pass, g_pass + g_fail);
    return (g_fail == 0) ? 0 : 1;
}
